test/charm_test.py

```diff
@@ -110,6 +76,95 @@ def test__init__works_without_a_hitch(self,
 #         # ]
 
 
+class OnConfigChangedHandlerTest(unittest.TestCase):
+
+    # We are mocking the time module here so that we don't actually wait
+    # 1 second per loop during test exectution.
+    @patch('charm.build_juju_unit_status', spec_set=True, autospec=True)
+    @patch('charm.k8s', spec_set=True, autospec=True)
+    @patch('charm.time', spec_set=True, autospec=True)
+    @patch('charm.build_juju_pod_spec', spec_set=True, autospec=True)
+    @patch('charm.set_juju_pod_spec', spec_set=True, autospec=True)
+    @patch('charm.StoredState', spec_set=True, autospec=True)
+    def test__it_blocks_until_pod_is_ready(
+            self,
+            mock_stored_state_cls,
+            mock_pod_spec,
+            mock_juju_pod_spec,
+            mock_time,
+            mock_k8s_mod,
+            mock_build_juju_unit_status_func):
+        # Setup
+        mock_fw_adapter_cls = \
+            create_autospec(framework.FrameworkAdapter, spec_set=True)
+        mock_fw_adapter = mock_fw_adapter_cls.return_value
+
+        mock_juju_unit_states = [
+            MaintenanceStatus(str(uuid4())),
+            MaintenanceStatus(str(uuid4())),
+            ActiveStatus(str(uuid4())),
+        ]
+        mock_build_juju_unit_status_func.side_effect = mock_juju_unit_states
+
+        mock_event_cls = create_autospec(EventBase, spec_set=True)
+        mock_event = mock_event_cls.return_value
+
+        mock_state = mock_stored_state_cls.return_value
+
+        # Exercise
+        charm.on_config_changed_handler(mock_event,
```

Comment:
Shouldn't we get away from `charm` object direct access? See https://github.com/relaxdiego/charm-k8s-prometheus/issues/18 for details - but tl;dr this can and will interfere with Harness tests. I'd suggest to re-work these tests using Harness or, at least, put it as a high-priority roadmap item after landing this PR to unblock the functionality.

Reply:
Note that I'm not actually calling a `charm.Charm` instance here. Rather I'm calling the stateless `on_config_changed_handler` function in the `charm` module.

Reply:
Ok, let's leave it as is now.